src/handler/core.rs

```diff
@@ -108,12 +171,63 @@ where
     pub fn chain(self, next: Self) -> Self {
         let required_update_kinds_set = self.description().merge_chain(next.description());
 
-        from_fn_with_description(required_update_kinds_set, move |event, cont| {
-            let this = self.clone();
-            let next = next.clone();
-
-            this.execute(event, |event| next.execute(event, cont))
-        })
+        let new_sig = match (&self.data.sig, &next.data.sig) {
+            (_, HandlerSignature::Entry) => {
+                panic!("Ill-typed handler chain: the second handler cannot be an entry")
```

Comment:
Why? `entry` as rhs should be trivially fine.

Reply:
Answered in https://github.com/teloxide/dptree/pull/23#discussion_r1199691364.